File: fs/numbers/src/NumbersRecursive.fs
namespace src

module NumbersRecursive =

    let rec fibonacci n =
        match n with
        | n when n <= 1 -> n
        | _ -> fibonacci (n - 1) + fibonacci (n - 2)

    let rec factorial n =
        match n with
        | 1 -> 1
        | n -> n * factorial (n - 1)

    let rec sum_numbers n =
        match n with
        | 0 -> 0
        | n -> n + sum_numbers (n - 1)

File: _experimental/_java/numbers/src/main/java/com/numbers/NumbersRecursive.java
public class NumbersRecursive {
    public int fibonacci(int n) {
        if (n <= 1) {
            return n;
        }
        return fibonacci(n - 1) + fibonacci(n - 2);
    }

    public int factorial(int n) {
        if (n == 0) {
            return 1;
        }
        return n * factorial(n - 1);
    }

    public int sumNumbers(int n) {
        if (n == 0) {
            return 0;
        }
        return n + sumNumbers(n - 1);
    }
}
